Add some basic auto routes

ptavolette opened the pull request:
- Changed assignments for Positions 1-3 and Game pieces 1-4 (Reversed to match coordinate order)
- Used coordinates pulled from CAD to change and add coordinates to be more accurate
- Added initial routes for driving to game pieces, returning from those pieces, from all 3 positions

DevilTech committed:
WIP auto routes

Doug-Tavolette committed:
Added inital routes for starting positions 1, 2, 3 - CW

MathNerd28 commented:
Correct me if I’m wrong, but don’t the indices of the Driver Stations go in the original direction? I’m just worried it’s going to get confusing during competitions. It would be easier to say “We’re at station 3, so let’s start auto in position 3”.

MathNerd28 commented on src/main/java/frc/robot/AutoRoutes.java:
```diff
@@ -11,116 +11,232 @@
-            new Thread(AutoRoutes::simulateTrajectories).start();
+        public static final double FIELD_LENGTH = 16.54;
+        public static final double FIELD_WIDTH  = 8.02;
+
```
Please run the formatter on this file

MathNerd28 commented on src/main/java/frc/robot/AutoRoutes.java:
```diff
@@ -11,116 +11,232 @@
+        private static final Pose2d S1_P1_E = new Pose2d(6.82, 0.92,
+                        DEGREES_180);
+
+        public static final Pose2d[] START_1_TO_PIECE_1 = { S1_P1_A, S1_P1_B,
+                        S1_P1_C, S1_P1_D, S1_P1_E };
```
These arrays must also include the start and end position, or else the robot will be very confused at the start!

MathNerd28 replied:
This applies to every array in this file

Cameron-Williams1 replied:
We removed those as it looked like it in your code it was referencing these positions. "public static final Pose2d START_POINT_1 = new Pose2d(2.2, 4.7, DEGREES_180);
    public static final Pose2d START_POINT_3 = new Pose2d(2.2, 4.7, DEGREES_180);
    public static final Pose2d START_POINT_2 = new Pose2d(2.2, 2.75, DEGREES_180);
    public static final Pose2d START_POINT_3 = new Pose2d(2.2, 0.75, DEGREES_180);" 
As S1_P1_A is our starting position we kept it out

MathNerd28 replied:
That’s a duplication of the starting position; let’s give the common poses more clear names than “S1_P1_A”, and save those for intermediate poses we don’t care about.

MathNerd28 commented on src/main/java/frc/robot/AutoRoutes.java:
```diff
@@ -11,116 +11,232 @@
+        public static final Pose2d[] START_2_OVER_CHARGE_STATION = { S2_OCS_A,
+                        S2_OCS_B, S2_OCS_C, S2_OCS_D, S2_OCS_E };
+
+        // public static final Pose2d[] START_1_TO_PIECE_1 = { START_POINT_1,
+        // S1_P1_A, S1_P1_B, S1_P1_C, S1_P1_D, S1_P1_E,
+        // GAME_PIECE_1 };
+        // public static final Pose2d[] PIECE_1_TO_START_1 = { GAME_PIECE_1,
+        // S1_P1_E, S1_P1_D, S1_P1_C, S1_P1_B, S1_P1_A,
+        // START_POINT_1 };
+
+        // private static final Pose2d S3_P4_A = new Pose2d(5.2, 0.75,
+        // DEGREES_180);
+        // private static final Pose2d S3_P4_B = new Pose2d(5.4, 0.85,
+        // DEGREES_150);
+        // private static final Pose2d S3_P4_C = new Pose2d(5.7, 0.9,
+        // DEGREES_90);
+        // private static final Pose2d S3_P4_D = new Pose2d(6.0, 0.9,
+        // DEGREES_30);
+        // private static final Pose2d S3_P4_E = new Pose2d(6.3, 0.9,
+        // DEGREES_0);
+
+        // public static final Pose2d[] PIECE_4_TO_START_3 = { GAME_PIECE_4,
+        // S3_P4_E, S3_P4_D, S3_P4_C, S3_P4_B, S3_P4_A,
+        // START_POINT_3 };
```
If these definitions are no longer valid, please delete them rather than just commenting them out.

MathNerd28 committed:
Merge branch 'master' into auto-routes

Doug-Tavolette committed:
Changed names to fit naming conventions

Doug-Tavolette committed:
Merge branch 'auto-routes' of https://github.com/Team1559/2023-Charged-Up into auto-routes

Doug-Tavolette committed:
Merge branch 'master' into auto-routes

Doug-Tavolette committed:
Merged master in branch and formatted AutoRoutes.java file

Cameron-Williams1 committed:
Work in progress

Cameron-Williams1 committed:
merge upstream

Cameron-Williams1 committed:
Working in progress

mmarchetti committed:
prep for auto route testing

JohnBurdick1 committed:
Merge branch 'master' into auto-routes

JohnBurdick1 committed:
Merge branch 'master' into auto-routes

MathNerd28 committed:
Merge branch 'master' into auto-routes

JohnBurdick1 committed:
Merge branch 'auto-routes' of https://github.com/Team1559/2023-Charged-Up into auto-routes

JohnBurdick1 committed:
Auto route scoring implementation

-Added some new auto routes that account for piece scoring, arm movement
-DO NOT USE "BLUE_SCORE_CUBE_PICKUP_CUBE_RETURN_TO_1". I DON'T KNOW IF THIS COMMAND WILL WORK.

MathNerd28 committed:
Add score commands "library" + implement automatic trajectory mirroring

MathNerd28 committed:
Actually call the selected autonomous command

JohnBurdick1 committed:
Merge branch 'master' into auto-routes

JohnBurdick1 committed:
Commit, added some auto routes

-Added some auto routes/commands to account for cones
-updated WPILib build
changed camera offset constants

MathNerd28 committed:
Merge branch 'master' into auto-routes

JohnBurdick1 committed:
minor position changes in Arm

JohnBurdick1 committed:
Merge branch 'auto-routes' of https://github.com/Team1559/2023-Charged-Up into auto-routes

MathNerd28 committed:
New auto routes, vision null checks, travel before auto, accel limits in all directions

MathNerd28 committed:
Double teleop wrist velocity

JohnBurdick1 committed:
Merge branch 'master' into auto-routes

MathNerd28 committed:
Merge branch 'master' into auto-routes

JohnBurdick1 committed:
Merge branch 'auto-routes' of https://github.com/Team1559/2023-Charged-Up into auto-routes

MathNerd28 committed:
Add field relative toggle, AlignToGrid command, and enable rPID swerve deadband

JohnBurdick1 committed:
Changes in "Robot" class, added SmartDashboard things

-added a CommandScheduler.getInstance.run() method into autonomousPeriodic and teleopPeriodic, removed from robotPeriodic
-added motor temp readings in ArmSegment

DevilTech committed:
wip updates

DevilTech committed:
Commit for 3/11/23

-Added new auto route for cone scoring
-Changed speed constants for wrist servo, added "med-mode" to swerve drive

DevilTech committed:
getrobotangle

mmarchetti committed:
motor init delay

mmarchetti committed:
Merge branch 'auto-routes' of github.com:Team1559/2023-Charged-Up into auto-routes

DevilTech committed:
driving fixes

MathNerd28 committed:
Add checks for accurate pose estimation before execution

MathNerd28 committed:
More logging + cancel lots of stuff

MathNerd28 committed:
New + adjusted auto routes

SOME ROUTES ARE INVALID, these are okay:
- Null
- Leave (1 and 3)
- Score cone, stay
- Score cone, leave (1 and 3)
